x-pack/plugins/apm/server/routes/services/get_services/get_service_aggregated_transaction_stats.ts

```diff
@@ -0,0 +1,163 @@
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { kqlQuery, rangeQuery } from '@kbn/observability-plugin/server';
+import { ProcessorEvent } from '@kbn/observability-plugin/common';
+import {
+  AGENT_NAME,
+  SERVICE_ENVIRONMENT,
+  SERVICE_NAME,
+  TRANSACTION_TYPE,
+  TRANSACTION_DURATION_SUMMARY,
+  TRANSACTION_FAILURE_COUNT,
+  TRANSACTION_SUCCESS_COUNT,
+} from '../../../../common/elasticsearch_fieldnames';
+import {
+  TRANSACTION_PAGE_LOAD,
+  TRANSACTION_REQUEST,
+} from '../../../../common/transaction_types';
+import { environmentQuery } from '../../../../common/utils/environment_query';
+import { AgentName } from '../../../../typings/es_schemas/ui/fields/agent';
+import { calculateThroughputWithRange } from '../../../lib/helpers/calculate_throughput';
+import { calculateFailedTransactionRateFromServiceMetrics } from '../../../lib/helpers/transaction_error_rate';
+import { ServicesItemsSetup } from './get_services_items';
+import { serviceGroupQuery } from '../../../../common/utils/service_group_query';
+import { ServiceGroup } from '../../../../common/service_groups';
+import { RandomSampler } from '../../../lib/helpers/get_random_sampler';
+
+interface AggregationParams {
+  environment: string;
+  kuery: string;
+  setup: ServicesItemsSetup;
+  maxNumServices: number;
+  start: number;
+  end: number;
+  serviceGroup: ServiceGroup | null;
+  randomSampler: RandomSampler;
+}
+
+export async function getServiceAggregatedTransactionStats({
+  environment,
+  kuery,
+  setup,
+  maxNumServices,
+  start,
+  end,
+  serviceGroup,
+  randomSampler,
+}: AggregationParams) {
+  const { apmEventClient } = setup;
+
+  const response = await apmEventClient.search(
+    'get_service_aggregated_transaction_stats',
+    {
+      apm: {
+        events: [ProcessorEvent.metric],
+      },
+      body: {
+        size: 0,
+        query: {
+          bool: {
+            filter: [
+              ...[{ exists: { field: TRANSACTION_DURATION_SUMMARY } }],
+              ...rangeQuery(start, end),
+              ...environmentQuery(environment),
+              ...kqlQuery(kuery),
+              ...serviceGroupQuery(serviceGroup),
+            ],
+          },
+        },
+        aggs: {
+          sample: {
+            random_sampler: randomSampler,
+            aggs: {
+              services: {
+                terms: {
+                  field: SERVICE_NAME,
+                  size: maxNumServices,
+                },
+                aggs: {
+                  transactionType: {
+                    terms: {
+                      field: TRANSACTION_TYPE,
+                    },
+                    aggs: {
+                      avg_duration: {
+                        avg: {
+                          field: TRANSACTION_DURATION_SUMMARY,
+                        },
+                      },
+                      total_doc: {
+                        value_count: {
```

Comment:
What kind of field type is transaction.duration.summary? Is it an aggregate metric double? If so, why is the value count agg necessary? Are we not using `_doc_count` or is that not supported for aggregate metric double?

Reply:
Yes, based on the [mapping](https://github.com/elastic/apm-server/pull/8607/files#diff-7308b0486f2df02c497f6063fd47c6e45078484b99c0d6deb635cdd22c90c9ccR21) is `aggregate_metric_double`

Reply:
Can you look into `_doc_count` (it is different from `doc_count`, it's metadata for a document indexed into ES)

Reply:
`_doc_count` and `value_count` agg should be the same? 

```
  {
            "key": "web-go-172",
            "doc_count": 11,
            "transactionType": {
              "doc_count_error_upper_bound": 0,
              "sum_other_doc_count": 0,
              "buckets": [
                {
                  "key": "request",
                  "doc_count": 11,
                  "total_doc": {
                    "value": 7854
                  },
                  .
                  .
}
```